google/cloud/internal/curl_impl.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_CURL_IMPL_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_CURL_IMPL_H
+
+#include "google/cloud/internal/curl_handle.h"
+#include "google/cloud/internal/curl_handle_factory.h"
+#include "google/cloud/internal/curl_wrappers.h"
+#include "google/cloud/internal/rest_request.h"
+#include "google/cloud/internal/rest_response.h"
+#include "google/cloud/options.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+// This class encapsulates use of libcurl and manages all the necessary state
```

Comment:
Looks like this is a placeholder so you can have an interface to code to while making progress on other things?  Maybe a TODO and a bug would be nice if that is the case.